clippy_lints/src/if_let_mutex.rs

```diff
@@ -63,18 +56,28 @@ impl<'tcx> LateLintPass<'tcx> for IfLetMutex {
         }) = higher::IfLet::hir(cx, expr)
         {
             op_visit.visit_expr(let_expr);
-            if op_visit.mutex_lock_called {
+            if let Some(op_mutex) = op_visit.found_mutex {
                 arm_visit.visit_expr(if_then);
                 arm_visit.visit_expr(if_else);
 
-                if arm_visit.mutex_lock_called && arm_visit.same_mutex(cx, op_visit.found_mutex.unwrap()) {
-                    span_lint_and_help(
+                if let Some(arm_mutex) = arm_visit.found_mutex_if_same_as(op_mutex) {
+                    let diag = |diag: &mut Diagnostic| {
+                        diag.span_label(
+                            op_mutex.span,
+                            "This Mutex will remain locked for the entire `if let`-block...",
```

Comment:
Nice improvements, only one tiny nit and then we can merge this. All diagnostic messages should start with lowercase letters:

```suggestion
                            "this Mutex will remain locked for the entire `if let`-block...",
```

Reply:
@xFrednet I'm not exactly familiar with the distinction: Are the `this mutex will remain...` messages `span_label` or `span_note`? The `span_label` currently used is visually less prominent, yet I'd rather go with `span_note`

Reply:
The guideline, that they should start with a lower letter, applies to all of them. In this case, all that needs to be done is change the first letter and bless the tests again.

Now generally, they are distinguished by what kind of message you want to add. This is my understanding of them:
* `span_label` will display the message and span in the main window. This is intended to show the root problem.
* `span_note` is intended for additional context or showcasing possible connected code
* `span_help`/`span_sugg` for messages with direct help and suggestions

For this and the second message, I believe that either `span_label` or `span_note` is good. I'm fine with this version, but you can try around with `span_note` to see if the message is then even clearer :upside_down_face: